File: app-cpp/src/ServerProcessingEngine.cpp
#include "ServerProcessingEngine.hpp"

ServerProcessingEngine::ServerProcessingEngine(std::shared_ptr<IndexStore> store) : store(store) {
    // TO-DO initialize FileRetrievalEngineImpl and gRPC server objects
}

void ServerProcessingEngine::initialize(int serverPort) {
    // TO-DO create and start the gRPC server thread
}

void ServerProcessingEngine::rungRPCServer(int serverPort) {
    // TO-DO build the gRPC server
    // TO-DO register the FileRetrievalEngineImpl service with the gRPC server
    // TO-DO start the gRPC server
}

void ServerProcessingEngine::shutdown() {
    // TO-DO shutdown the gRPC server
    // TO-DO wait for the gRPC server to shutdown
    // TO-DO join the gRPC server thread
}

File: app-java/src/main/java/csc435/app/ServerProcessingEngine.java
package csc435.app;

import io.grpc.Grpc;
import io.grpc.InsecureServerCredentials;
import io.grpc.ServerBuilder;

public class ServerProcessingEngine {
    private IndexStore store;
    // TO-DO keep track of the RPCServerWorker object
    // TO-DO keep track of the gRPC server thread

    public ServerProcessingEngine(IndexStore store) {
        this.store = store;
    }

    // TO-DO create and start the gRPC Server
    public void initialize(int serverPort) {
        // TO-DO create the RPCServerWorker object
        // TO-DO create and start the gRPC server thread that runs in the context of the RPCServerWorker object
    }

    // TO-DO shutdown the gRPC Server
    public void shutdown() {
        // TO-DO call shutdown on the RPCServerWorker object
        // TO-DO join the gRPC server thread
    }
}
